python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -204,6 +204,22 @@ class NearestNeighbors(Base):
         self.n_neighbors = n_neighbors
         self._should_downcast = should_downcast
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        del state['handle']
+        state['X_m'] = cudf.DataFrame.from_gpu_matrix(self.X_m)
+
+        return state
+
+    def __setstate__(self, state):
+        super(NearestNeighbors, self).__init__(handle=None,
+                                               verbose=state['verbose'])
```

Comment:
is 'verbose' guaranteed to exist in `state`? not sure if it needs to be complete or not by definition.

Reply:
yes, it is inherited from `Base`